### DISCUSSION TITLE
Limit People search to specific SharePoint Group

### DISCUSSION THREAD

**Discussion by Lennart28213** [microsoft-search/pnp-modern-search#2348] [Q&A]
Hi, 

Is it possible to limit the PNP search to a specific Sharepoint group instead of showing the entire organization? I have already tried to do this with "path:XXX" but have had no luck so far.
Do you have any tips in this regard?

Thank you very much and best regards.

Lennart

**Comment by wobba** [comment]
Can you define what you mean with: "SharePoint group"?

All of these should work in the query template, and also check https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#connections-tokens for usage of contextual tokens:

* `path:https://tenant.sharepoint.com/sites/foo/` <- limit to one site
* `DepartmentId:<guid of hub site>`
* `SiteID:<site id of the site>`
* `GroupId:<guid of the O365 group>`

**Comment by Lennart28213** [comment]
Hello Wobba,

 thank you for your reply. Please let me clarify my request.

My organization consists of almost 2000 employees and is organized in departments like marketing or business development. These departments have their own sharepoint member groups. For example, the marketing department might have 45 employees in their group. 
![image](https://user-images.githubusercontent.com/105660404/189607169-a5fee29e-f262-414b-b1b7-3a8c3f49b80b.png)

Through these groups we manage permissions for files and site collections. Now a group/department wants to have its own employee directory and we want to implement this using PNP search. As seen in this screenshot:

![image](https://user-images.githubusercontent.com/105660404/189607500-92ca7ec5-a6a7-4324-b74f-ee086f9c9e22.png)


Now here we want to restrict the employees in this directory to a specific sharepoint group (e.g. Marketing). I have tried to access the corresponding group via the query template using the "path" command, but have had no success so far.

![image](https://user-images.githubusercontent.com/105660404/189607612-e30ca345-dd24-471f-8751-189f4220999f.png)

Based on your first answer, I assume I need to use "GroupId:<guid of the O365 group>>" to achieve the desired result?

Thank you very much and best regards.

**Reply by wobba** [reply]
There is no way to target SharePoint membership groups or any access control via search. Search always return what you have read access to, so if the groups are mutual exclusive in terms of what you can read/access, then no special targeting is needed. If access control goes outside of the groups you want to target with, then you need to come up with some other solution.

**Reply by Lennart28213** [reply]
Hi Wobba,

thanks for your reply. I will then have to look for another solution.
best regards;

**Reply by DanRuz54** [reply]
@Lennart28213 could you share your solution for this? Do you have any? I try to solve same problem. 

Thx. 

Dan

**Reply by wobba** [reply]
You need to use profile properties for targeting in search. `GroupId` will work, if the site is tied to a M365 Group (AAD) via Outlook Groups, Yammer Groups or a Team.
